### PR TITLE
Analyze dependencies in rmarkdown parameter expressions

### PR DESCRIPTION
**Pull request by aronatkins**
Depends on https://github.com/yihui/knitr/pull/1078

### CODE REVIEW

File: R/dependencies.R [hunk 1]
```diff
@@ -146,18 +143,35 @@ yamlDeps <- function(yaml) {
...
+    yaml <- content[(start + 1):(end - 1)]
+    yaml <- paste(yaml, collapse = "\n")
+    deps <- append(deps, yamlDeps(yaml))
+
+    if (requireNamespace("knitr", quietly = TRUE)) {
+      if (packageVersion("knitr") >= "1.10.18") {
+        knit_params <- knitr::knit_params_yaml(yaml, evaluate = FALSE)
+        if (length(knit_params) > 0) {
+          # Parameterized R Markdown documents can use a Shiny customization app.
+          deps <- append(deps, "shiny")
```
Comment (kevinushey) [inline R/dependencies.R]: Does this imply that any `.Rmd` containing custom parameters has `shiny` as an implicit dependency?

Comment (aronatkins) [inline R/dependencies.R]: Yes. The application will be embedded in `rmarkdown`.

Comment (aronatkins) [inline R/dependencies.R]: That said, we basically dup the `shiny` dependency injection in my `rsconnect` PR, but I think the long-term view is that `rsconnect` should get out of the business of doing its own dependency analysis.

**Comment by kevinushey** [comment]
This looks good to me and it looks like the dependant PR has now been merged -- is this safe to take?

**Comment by aronatkins** [comment]
I think it's good. I'm on vacation for the next week and a half. @jmcphers I have a similar set of changes for `rsconnect` (which uses this `packrat` PR in some code paths). You OK with this change?

**Comment by jmcphers** [comment]
Yes, this looks good to me.

**Comment by kevinushey** [comment]
It looks like conflicts were introduced following the recent YAML change -- I can fix this PR up and merge later today.

**Comment by kevinushey** [comment]
'Merged' with minor changes in commit here: https://github.com/rstudio/packrat/commit/dfd1c904a0feb74422233f393cc324daf6517ea0

**Pull request closed without merging**